pkg/controllers/customroute/custom_route_controller.go

```diff
@@ -0,0 +1,509 @@
+	// update ingressConfig status
+	existingRoute, err := c.routeLister.Routes("openshift-authentication").Get("oauth-openshift")
+	if err != nil {
+		return err
+	}
+
+	_, err = c.updateComponentRouteStatus(ctx, &configv1.ComponentRouteStatus{
+		Namespace:        c.componentRoute.Namespace,
+		Name:             c.componentRoute.Name,
+		DefaultHostname:  configv1.Hostname("oauth-openshift." + ingress.Spec.Domain),
+		CurrentHostnames: []configv1.Hostname{configv1.Hostname(existingRoute.Spec.Host)},
+		ConsumingUsers: []configv1.ConsumingUser{
+			"system:serviceaccount:oauth-openshift:authentication-operator",
+		},
+		RelatedObjects: []configv1.ObjectReference{
+			{
+				Group:     routev1.GroupName,
+				Resource:  "routes",
+				Name:      "oauth-openshift",
+				Namespace: "openshift-authentication"},
+		},
+		Conditions: c.getConditions(customRouteErrors, existingRoute, ingress),
+	})
+	return err
+}
+
+func (c *customRouteController) getConditions(errs []error, route *routev1.Route, ingress *configv1.Ingress) []metav1.Condition {
+	now := metav1.Now()
+	if len(errs) != 0 {
+		return []metav1.Condition{
+			{
+				LastTransitionTime: now,
+				Type:               "Degraded",
+				Status:             metav1.ConditionTrue,
+				Reason:             "CustomRouteError",
+				Message:            fmt.Sprintf("Error Configuring custom route: %v", errs),
+			},
+			{
+				LastTransitionTime: now,
+				Type:               "Progressing",
+				Status:             metav1.ConditionFalse,
+				Reason:             "CustomRouteError",
+				Message:            fmt.Sprintf("Error Configuring custom route: %v", errs),
+			},
+		}
+	}
+
+	if _, _, err := routeapihelpers.IngressURI(route, route.Spec.Host); err != nil {
+		return []metav1.Condition{
+			{
+				LastTransitionTime: now,
+				Type:               "Degraded",
+				Status:             metav1.ConditionTrue,
+				Reason:             "RouteNotAdmitted",
+				Message:            fmt.Sprintf("Route not admitted: %v", err),
+			},
+			{
+				LastTransitionTime: now,
+				Type:               "Progressing",
+				Status:             metav1.ConditionFalse,
+				Reason:             "RouteNotAdmitted",
+				Message:            fmt.Sprintf("Route not admitted: %v", err),
+			},
+		}
+	}
+
+	if err := c.checkRouteAvailable(route.Spec.Host, ingress); err != nil {
+		return []metav1.Condition{
+			{
+				LastTransitionTime: now,
+				Type:               "Degraded",
+				Status:             metav1.ConditionFalse,
+				Reason:             "ErrorReachingOutToService",
+				Message:            fmt.Sprintf("unexpected error at %s: %v", route.Spec.Host, err),
+			},
+			{
+				LastTransitionTime: now,
+				Type:               "Progressing",
+				Status:             metav1.ConditionTrue,
+				Reason:             "ErrorReachingOutToService",
+				Message:            fmt.Sprintf("unexpected error at %s: %v", route.Spec.Host, err),
+			},
```

Comment:
why is this degraded=false,progressing=true?

Reply:
Good question - I had assumed that the admitted route might eventually become available. Instead, we may want to follow your suggestion [here](https://github.com/openshift/cluster-authentication-operator/pull/430#discussion_r623079557) and report:
* Degraded = false and progressing = true  if less than some time has passed (say 5 minutes)
* Degraded = true and progressing = false  if 5 minutes has passed